Northstar.Client/mod/scripts/vscripts/ui/custom_scrollbar.nut

```diff
@@ -0,0 +1,209 @@
+{
+	if( FindScrollbarSafe( scrollbar ) )
+	{
+		printt( format( "%s is already registered. omitting registration.", scrollbar.tostring() ) )
+		return
+	}
+
+	var cover = Hud_GetChild( scrollbar, "SliderCover" )
+	AddButtonEventHandler( cover, UIE_GET_FOCUS, CoverGetFocus )
+	file.covers.append( cover )
+	ScrollbarExt bar
+	bar.originalX = expect int( Hud_GetPos( cover )[0] )
+	bar.originalY = expect int( Hud_GetPos( cover )[1] )
+	bar.originalHeight = Hud_GetHeight( scrollbar )
+	bar.originalWidth = Hud_GetWidth( scrollbar )
+	bar.scrollbar = scrollbar
+	bar.cover = cover
+	bar.movementCapture = Hud_GetChild( scrollbar, "MouseMovementCapture" )
+	bar.button = Hud_GetChild( scrollbar, "SliderButton" )
+	bar.panel = Hud_GetChild( scrollbar, "BtnModListSliderPanel" )
+	bar.horizontal = horizontal
+	file.scrollbars.append( bar )
+}
+
+ScrollbarExt ornull function FindScrollbarSafe( var scrollbar )
+{
+	foreach( ScrollbarExt container in file.scrollbars )
+		if( container.scrollbar == scrollbar )
+			return container
+	return null
+}
+
+ScrollbarExt function FindScrollbar( var scrollbar )
+{
+	ScrollbarExt ornull scr = FindScrollbarSafe( scrollbar )
+	if( scr != null )
+		return expect ScrollbarExt( scr )
+	throw "scrollbar not registered"
+	unreachable
+}
+
+// * #####################
+// * internal functions ##
+// * #####################
+
+void function CoverGetFocus( var cover )
+{
+	Hud_SetVisible( cover, false )
+	if( file.invisCover )
+		Hud_SetVisible( file.invisCover, true )
+	file.invisCover = cover
+
+	ScrollbarExt s = file.scrollbars[ GetFocusedScrollbarIndex() ]
+	Hud_SetFocused( s.button )
+}
+
+void function MouseMovementHandler( int x, int y )
+{
+	int idx = GetFocusedScrollbarIndex()
+	Hud_SetFocused( file.scrollbars[ idx ].button )
+	if( idx < 0 )
+		return
+	foreach( void functionref( int x, int y ) callback in file.scrollbars[ idx ].callbacks )
+	{
+		callback( x, y )
+	}
+
+	if( file.scrollbars[ idx ].horizontal )
+		RepositionScrollbar_Horizontal( file.scrollbars[ idx ], x, y )
+	else
+		RepositionScrollbar_Vertical( file.scrollbars[ idx ], x, y )
+}
+
+void function RepositionScrollbar_Horizontal( ScrollbarExt scrollbar, int x, int y )
+{
+	x = x * -1
+	y = y * -1
+
+	int minXPos = scrollbar.originalX
+	int maxXPos = scrollbar.originalX + scrollbar.originalWidth - Hud_GetWidth( scrollbar.button )
+
+	int pos = expect int( Hud_GetPos( scrollbar.button )[0] )
+	int newPos = pos - x
+
+	if ( newPos > maxXPos ) newPos = maxXPos
+	if ( newPos < minXPos ) newPos = minXPos
+
+	int yPos = expect int( Hud_GetPos( scrollbar.button )[1] )
+
+	Hud_SetPos( scrollbar.cover, newPos, yPos )
+	Hud_SetPos( scrollbar.movementCapture, newPos, yPos )
+	Hud_SetPos( scrollbar.button, newPos, yPos )
+	Hud_SetPos( scrollbar.panel, newPos, yPos )
+}
+
+void function RepositionScrollbar_Vertical( ScrollbarExt scrollbar, int x, int y )
+{
+	x = x * -1
+	y = y * -1
+
+	int[2] screenSize = GetScreenSize()

```

Comment:
This is unused.
Havent messed with this since Better.ServerBrowser was merged, but have you tested other resolutions / aspect ratios ? Have a strong feeling this may break;